From: Alexander Y. Fomichev <git.user@gmail.com>
Date: Fri, 26 Oct 2007 10:05:57 +0000 (+0400)
Subject: [PATCH] Fix compilation issues in vpid code
X-Git-Tag: 028stab048~2
X-Git-Url: http://git.openvz.org/?p=linux-2.6.18-openvz;a=commitdiff_plain;h=0a17273c35f9fd043979da9ef216305f07ef369e

[PATCH] Fix compilation issues in vpid code
Patch fixes compilation issue in kernel/pid.c:
...
kernel/pid.c: In function 'free_pid':
kernel/pid.c:197: error: dereferencing pointer to incomplete type
kernel/pid.c: In function 'alloc_pid':
kernel/pid.c:233: error: dereferencing pointer to incomplete type
make[2]: *** [kernel/pid.o] Error 1
make[1]: *** [kernel] Error 2

If linux/kmem_cache is not included kmem_cache type is undefined.

http://bugzilla.openvz.org/show_bug.cgi?id=689
---

diff --git a/kernel/pid.c b/kernel/pid.c
index c6678a6..9d5208d 100644
--- a/kernel/pid.c
+++ b/kernel/pid.c
@@ -22,6 +22,7 @@
 
 #include <linux/mm.h>
 #include <linux/module.h>
+#include <linux/kmem_cache.h>
 #include <linux/slab.h>
 #include <linux/init.h>
 #include <linux/bootmem.h>
